src/main.rs

```diff
@@ -57,37 +57,38 @@ fn process_arguments(size: TerminalSize, args: Arguments) -> Result<(), Box<dyn
         println!("Terminal: {}", args.terminal_capabilities.name);
         Ok(())
     } else {
-        let (base_dir, input) = read_input(&args.filename)?;
-        let mut options = Options::empty();
-        options.insert(Options::ENABLE_TASKLISTS);
-        options.insert(Options::ENABLE_STRIKETHROUGH);
-        let parser = Parser::new_ext(&input, options);
-
-        if args.dump_events {
-            mdcat::dump_events(&mut std::io::stdout(), parser)?;
-            Ok(())
-        } else {
-            let syntax_set = SyntaxSet::load_defaults_newlines();
-            mdcat::push_tty(
-                &mut stdout(),
-                args.terminal_capabilities,
-                TerminalSize {
-                    width: args.columns,
-                    ..size
-                },
-                parser,
-                &base_dir,
-                args.resource_access,
-                syntax_set,
-            )?;
-            Ok(())
+        for filename in args.filenames {
+            let (base_dir, input) = read_input(filename)?;
+            let mut options = Options::empty();
+            options.insert(Options::ENABLE_TASKLISTS);
+            options.insert(Options::ENABLE_STRIKETHROUGH);
+            let parser = Parser::new_ext(&input, options);
+
+            if args.dump_events {
+                mdcat::dump_events(&mut std::io::stdout(), parser)?;
+            } else {
+                let syntax_set = SyntaxSet::load_defaults_newlines();
+                mdcat::push_tty(
+                    &mut stdout(),
+                    &args.terminal_capabilities,
+                    TerminalSize {
+                        width: args.columns,
+                        ..size
+                    },
+                    parser,
+                    &base_dir,
+                    args.resource_access,
+                    syntax_set,
+                )?;
```

Comment:
Does this work?  Does the question mark here acutally fail and exit the loop if `push_tty` fails?  Sorry if that's a stupid question but I'm not so familiar with Rust.

This actually raises another question: What's your intended error behaviour? Assume `mdcat foo bar` fails to read `foo`, would you like it to abort or try continue with `bar`?  

I guess I'd prefer the latter behaviour, but then you'd need to either handle errors right here, or accumulate all errors and return a list of errors from this function.

If you prefer the former behaviour please do test whether it behaves correctly currently, i.e. if mdcat prints the error properly if used with a non-existing file.

Reply:
The mdcat code currently exits with the error message.

**target/debug/mdcat CONTRIBUTING.md IDoNotExist CONTRIBUTING.md**
> ┄Contributing to mdcat
>
> Thank you for your interest in this little tool.  I appreciate all kinds of
> ...
>  2. Install cargo-release 0.9 or newer if not already present, with cargo install cargo-release.
>  3. Run cargo release and follow the instructions.
> **Error: No such file or directory (os error 2)**

Reply:
My intended behavior was to match what the code did before which was the exit on failure. 

I guess it depends on your belief should we "fail fast" or "fail-safe".  I am in the fail-fast camp since if a file is missing (ex. misspelled) you want to know it before moving on.  Why would you want to concatenate files that do not exist?   Also, the error message is written to standard error, so you might miss seeing the message if we follow the fail-safe philosophy.

Reply:
I just checked what `cat` does and it does it differently:  It doesn't exit on the first error, but just shows an error messages, keeps printing all files and eventuelly exists non-zero.  I guess that's "fail-safe" in your words, isn't it?

Let's follow that example. 

I also cam across the following example in the [`cat(1)`](https://man.openbsd.org/cat.1#EXAMPLES):

> Print the contents of file1, print data it receives from the standard input until it receives an EOF (‘^D’) character, print the contents of file2, read and output contents of the standard input again, then finally output the contents of file3. Note that if the standard input referred to a file, the second dash on the command line would have no effect, since the entire contents of the file would have already been read and printed by cat when it encountered the first ‘-’ operand.
>
>    $ cat file1 - file2 - file3

Can we get that to work for `mdcat` as well?

Reply:
Well, the good news the mulitple stdin case works as defined above.


> target/debug/mdcat CONTRIBUTING.md - CONTRIBUTING.md - CONTRIBUTING.md <stdin_file
> ┄Contributing to mdcat
> ...
> 3. Run cargo release and follow the instructions.
> **Data from stdin_data file**
> ┄Contributing to mdcat
> ...
> 3. Run cargo release and follow the instructions.
> ┄Contributing to mdcat
> ...
>  3. Run cargo release and follow the instructions.
> ➜